types/lodash/lodash-tests.ts

```diff
@@ -5306,13 +5306,29 @@ fp.now(); // $ExpectType number
     const defaultValue: boolean = anything;
     const objectWithOptionalField: { a?: boolean } = anything;
 
+    const anyNumber: number = anything;
+    const arrayOfNumbers: number[] = anything;
+    const tupleOfNumbers: [1] = anything;
+
+    const dictionary: _.Dictionary<string> = anything;
+    const maybeObject: { a: _.Dictionary<string> } | undefined = anything;
+
     _.get([], Symbol.iterator);
     _.get([], [Symbol.iterator]);
 
     _.get("abc", 1); // $ExpectType string
     _.get({ a: false }, "a"); // $ExpectType boolean
     _.get(objectWithOptionalField, "a"); // $ExpectType boolean | undefined
     _.get(objectWithOptionalField, "a" as string); // $ExpectType any
+    _.get({ a: arrayOfNumbers }, 'a.0'); // $ExpectType number
+    _.get({ a: arrayOfNumbers }, 'a[0]'); // $ExpectType number
+    _.get({ a: arrayOfNumbers }, `a[${anyNumber}]`); // $ExpectType number
```

Comment:
```
    _.get({ a: arrayOfNumbers }, 'a.0'); // $ExpectType number | undefined
    _.get({ a: arrayOfNumbers }, 'a[0]'); // $ExpectType number | undefined
    _.get({ a: arrayOfNumbers }, `a[${anyNumber}]`); // $ExpectType number | undefined
```

IMO array index access should include `undefined` (maybe conditional on whether `--noUncheckedIndexAccess` is enabled? is that possible?)

Reply:
The type is defined as `T[number]` so maybe that already works? I'm guessing that the tests only run with `--noUncheckedIndexAccess` disabeld; I don't think there's any way to enable it. I'm curious what happens if someone manually tests this in a project with `--noUncheckedIndexAccess` enabled though.

Reply:
I agree it would be nice to have it with enabled `noUncheckedIndexAccess` (but not otherwise), but seems TS only derives `number` for these two cases. Checked in the playground against 4.8.4 and nightly 5.0.0-dev.20221103
```
declare const firstValue: Array<number>[number]
declare const anyValue: Array<number>[number]
```

Reply:
I raised a bug about it, turns out this is by design in TS - it won't ever express `number[][number]` / `Array<number>[number]` as nullable for the foreseeable future.

This stupid hack can detect whether noUncheckedIndexAccess is enabled:
```ts
const a: string[] = [];
const b = a[0];
type IsNoUncheckedIndexAccessEnabled = typeof b extends NonNullable<typeof b> ? 0 : 1;
```

I'd argue to just include `undefined` regardless of the compiler flag though (seems like a bug to me if it doesn't include it when the flag is on), because more safety is always better.

Reply:
Anyway, such change is out of this PR scope. Here, I only wanted to fix the broken `a[${number}]` case that right now returns `never`.